### DISCUSSION TITLE
Upsert problem

### DISCUSSION THREAD

**Discussion by Tihomir971** [supabase/supabase#6361] [Questions] [answered]
When using data upsert:
const { data, error } = await supabase
  .from('users')
  .upsert({ username: 'supabot' }, { onConflict: 'username' })
my primary key (sequential unique number) is geting bigger.
I'm upserting every day about 2K rows (almost no new) and now my primary key is at 591756.

Any suggestion?

**Comment by GaryAustin1** [comment, marked as answer]
I think is a good explanation:
https://stackoverflow.com/questions/37204749/serial-in-postgres-is-being-increased-even-though-i-added-on-conflict-do-nothing
Basically it increments whether you do insert or not.  But it is only used on a new insert.

**Reply by wiesson** [reply]
Wow, I've always asked myself that question, thanks for the link
